hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCachePreloaderStressTest.java

```diff
@@ -0,0 +1,119 @@
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static junit.framework.TestCase.assertNull;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class ClientMapNearCachePreloaderStressTest extends HazelcastTestSupport {
+
+    private final TestHazelcastFactory factory = new TestHazelcastFactory();
+
+    @Before
+    public void setUp() throws Exception {
+
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        factory.shutdownAll();
+    }
+
+
+    @Test
+    public void testDestroyAndCreateProxyWithNearCache() throws InterruptedException {
+        factory.newHazelcastInstance();
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.addNearCacheConfig(getNearCacheConfig("test"));
+        final HazelcastInstance client = factory.newHazelcastClient(clientConfig);
+
+        int createPutGetThreadCount = 100;
+        int destroyThreadCount = 2;
+        ExecutorService pool = Executors.newFixedThreadPool(createPutGetThreadCount + destroyThreadCount);
+
+        final AtomicBoolean isRunning = new AtomicBoolean(true);
+        final AtomicReference<Exception> exception = new AtomicReference<Exception>();
+        for (int i = 0; i < destroyThreadCount; i++) {
+            pool.execute(new Runnable() {
+                @Override
+                public void run() {
+                    while (isRunning.get()) {
+                        for (DistributedObject distributedObject : client.getDistributedObjects())
+                            distributedObject.destroy();
+                    }
+                }
+            });
+        }
+
+        for (int i = 0; i < createPutGetThreadCount; i++) {
+            pool.execute(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        while (isRunning.get()) {
+                            IMap<Object, Object> map = client.getMap("test");
+                            map.put(1, 1);
+                            map.get(1);
+                        }
+                    } catch (Exception e) {
+                        isRunning.set(false);
+                        e.printStackTrace(System.out);
+                        exception.set(e);
+                    }
+                }
+            });
+        }
+
+        Thread.sleep(5000);
```

Comment:
We have `sleepSeconds(5)` via `HazelcastTestSupport`, which will possibly also get rid of the `throws InterruptedException`.